onebusaway-android/src/main/java/org/onebusaway/android/ui/TripPlanFragment.java

```diff
@@ -127,6 +136,45 @@ interface Listener {
 
     private FirebaseAnalytics mFirebaseAnalytics;
 
+    //sets the address in the text view from the user selected contact
+    private final String ID = "id";
+    private ActivityResultLauncher<TextView> mSelectAddressFromContactLauncher = registerForActivityResult(
```

Comment:
You can separate `ActivityResultContract` and `ActivityResultLauncher` as two classes for better readability also it can be like this:

```Java
private final ActivityResultContract<TextView, Intent> selectAddressFromContactContract = new ActivityResultContract<TextView, Intent>() {
    private int addressInputId;

    @NonNull
    @Override
    public Intent createIntent(@NonNull Context context, TextView textView) {
        Intent intent = new Intent(Intent.ACTION_PICK);
        intent.setType(ContactsContract.CommonDataKinds.StructuredPostal.CONTENT_TYPE);
        addressInputId = textView.getId();
        return intent;
    }

    @Override
    public Intent parseResult(int resultCode, @Nullable Intent intent) {
        if (intent != null) {
            return intent.putExtra(ID, tvId);
        }
        return null;
    }
};

private final ActivityResultLauncher<TextView> mSelectAddressFromContactLauncher = registerForActivityResult(
        selectAddressFromContactContract,
        addressIntent -> {
            if (addressIntent == null) {
                return;
            }
            Uri addressUri = addressIntent.getData();
            String[] projection = new String[]{FORMATTED_ADDRESS};
            try (Cursor cursor = getContext().getContentResolver().query(addressUri, projection, null, null, null)) {
                if (cursor != null && cursor.moveToFirst()) {
                    int addressIndex = cursor.getColumnIndex(ContactsContract.CommonDataKinds.StructuredPostal.FORMATTED_ADDRESS);
                    String address = cursor.getString(addressIndex).replace("\n", ", ");
                    TextView addressInputId = getActivity().findViewById(addressIntent.getIntExtra(ID, -1));
                    addressInputId.post(() -> tv.setText(address));
                }
            }
        }
);
```